fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/LogicalPlanTrinoBuilder.java

```diff
@@ -175,7 +176,8 @@ protected Expression visitCast(io.trino.sql.tree.Cast node, ParserContext contex
         Expression expr = visit(node.getExpression(), context, Expression.class);
         DataType dataType = mappingType(node.getType());
         Expression cast = new Cast(expr, dataType);
-        if (dataType.isStringLikeType() && ((CharacterType) dataType).getLen() >= 0) {
+        if (dataType.isStringLikeType() && ((CharacterType) dataType).getLen() >= 0
+                && !(dataType.isVarcharType() && ((VarcharType) dataType).isWildcardVarchar())) {
```

Comment:
add comment to explain why max length not need to convert to `substr`

Reply:
done